### PR TITLE
Added support for recursive pattern in dependency xml files

### PR DESCRIPTION
**Pull request by alexeykostin**
## What does this PR do?

Added support for recursive pattern in dependency xml files

One can use the pattern 

`foo/bar/**`

to add dependency for all assets under `foo/bar/` path hint.

Use 

`foo/bar/**/*.png`

to match all `*.png` assets under `foo/bar/` path hint.

## How was this PR tested?

Tested locally by adding / removing the recursive pattern from existing `_dependency.xml`'s and checking the resulting asset lists

### CODE REVIEW

**Comment by carb-lloyd** [comment]
@alexeykostin  - let's simplify this back to the way you had before, in that:
foo/bar/* --> non-recursive, everything in that folder
foo/bar/** --> recursive, everything in every child folder.

And remove the foo/bar/**/*.png support

If needed we can look at better regex etc support later, but it's not really worth the effort for now

**Review by carb-lloyd — CHANGES_REQUESTED** [review]
Requesting changes:
Remove `**/.*`
Keep support of `/**` and verify.  Create new pr in our project w/ update to our seedlists

File: Code/Tools/AssetProcessor/native/AssetManager/PathDependencyManager.cpp [hunk 1]
```diff
@@ -562,7 +566,11 @@ namespace AssetProcessor
                             //  2. When another product is created, all existing wildcard dependencies are compared against that product to
                             //  see if it matches them.
                             // This check here makes sure that the filter for 1 matches 2.
-                            if (!isExactDependency)
+                            if (!isExactDependency
+#if defined(CARBONATED)
+                                && !isRecursiveDependency
+#endif
+                                )
```
Comment (carb-lloyd) [inline Code/Tools/AssetProcessor/native/AssetManager/PathDependencyManager.cpp]: Small thing, but this makes diffs difficult during engine updates difficult.
Do this instead:
```
#if defined(CARBONATED)
if (!isExactDependency && isRecursiveDependency)
#else
if (!isExactDependency)
#endif
```

**Review by carb-lloyd — CHANGES_REQUESTED** [review]

**Review by carb-lloyd — APPROVED** [review]

**Comment by carb-lloyd** [comment]
@alexeykostin  - please merge and close when tested.

**Comment by alexeykostin** [comment]
Tested locally.

**Pull request merged**